cli/flox/src/commands/init.rs

```diff
@@ -262,6 +275,490 @@ impl InitHook for Requirements {
+            // the correct hook
+            if choice == 3 {
+                self.hook = Some(NodePackageManager::Npm)
+            } else if choice == 4 {
+                self.hook = Some(NodePackageManager::Yarn)
+            }
+            let message = formatdoc! {"
+
+                {}
+                Would you like Flox to apply one of these modifications?
+                You can always change the environment's manifest with 'flox edit'
+            ", format_customization(&self.get_init_customization())?};
+
+            let dialog = Dialog {
+                message: &message,
+                help_message: Some(AUTO_SETUP_HINT),
+                typed: Select {
+                    options: options.clone(),
+                },
+            };
+
+            (choice, _) = dialog.raw_prompt()?;
+        }
+
+        if choice == 0 {
+            self.hook = Some(NodePackageManager::Npm)
+        } else if choice == 1 {
+            self.hook = Some(NodePackageManager::Yarn)
+        }
+        Ok(choice == 0 || choice == 1)
+    }
+}
+
+impl InitHook for Node {
+    fn should_run(&mut self, _path: &Path) -> Result<bool> {
+        match self.get_action() {
+            NodeAction::Install(_) => {
+                debug!("Should run node init hook with requested nodejs version");
+                Ok(true)
+            },
+            NodeAction::OfferFloxDefault => {
+                debug!("Should run node init hook with default nodejs version");
+                Ok(true)
+            },
+            NodeAction::Nothing => {
+                debug!("Should not run node init hook");
+                Ok(false)
+            },
+        }
+    }
+
+    fn prompt_user(&mut self, _path: &Path, flox: &Flox) -> Result<bool> {
+        if let Some(NodePackageManager::Both) = self.hook {
+            self.prompt_for_hook(flox)
+        } else {
+            self.prompt_with_known_hook(flox)
+        }
+    }
+
+    fn get_init_customization(&self) -> InitCustomization {
+        let nodejs_to_install = match self.get_action() {
+            NodeAction::Install(result) => PackageToInstall {
+                id: "nodejs".to_string(),
+                pkg_path: result.rel_path.join("."),
+                version: result.version,
+                input: None,
+            },
+            NodeAction::OfferFloxDefault => PackageToInstall {
+                id: "nodejs".to_string(),
+                pkg_path: "nodejs".to_string(),
+                version: None,
+                input: None,
+            },
+            NodeAction::Nothing => unreachable!(),
+        };
+
+        let hook = match self.hook {
+            None => None,
+            Some(NodePackageManager::Npm) => Some("# TODO: npm hook".to_string()),
+            Some(NodePackageManager::Yarn) => Some("# TODO: yarn hook".to_string()),
+            // Default to npm
+            Some(NodePackageManager::Both) => Some("# TODO: npm hook".to_string()),
```

Comment:
`Both` representing _one of them_ is rather surprising

Reply:
I'll add some comments and maybe rename the enum. In this case `Both` means both lockfiles are present, but we decide to only add a hook for one of them